stats.py

```diff
@@ -1,17 +1,70 @@
 """Some simple statistical functions."""
 
-def mean(data):
-    """Takes the mean/average of a list of numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
-    pass
+import numbers
+import collections
+import math
 
-def median(numbers):
-    """Finds the median or middle of a list of sorted numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Median"""
-    pass
+def mean(number_list):
+    """Takes the mean/average of a list of numbers, a simple measure of 'middle'.
+    See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
+    #preconditions
+    assert isinstance(number_list, collections.Iterable), "Must be iterable."
```

Comment:
list can't be empty. you need at least one member. you'll get a divide by 0 on line 14 if you have an empty list.

assert list isn't empty. but you *also* don't need at least two members. that's what i was alluding to with an earlier comment.